sequence/sequences/compose/arith.py

```diff
@@ -0,0 +1,74 @@
+from sequence.core.core import Sequence
+
+import numpy as np
+
+
+class Arith(Sequence):
+    """Sequences combined via arithmetic operation, e.g., addition, multiplication.
+
+    Parameters
+    ----------
+    sequences : List[Sequence]
+        List of sequences to combine.
+    operation : None, str, function, or numpy ufunc, optional, default = None = mean
+        if str, must be one of "mean", "+" (add), "*" (multiply), "max", "min"
+        if func, must be of signature (1D iterable) -> float
+        operation carried out on the sequences
+    """
+
+    def __init__(self, sequences, operation):
+        super().__init__()
+        self.sequences = sequences
+        self.operation = operation
+        self._op = self._resolve_operation(operation)
+
+    def __add__(self, other):
+        if not isinstance(other, Arith) or self._op != other._op:
+            return Arith(sequences=[*self.sequences, other], operation="+")
+        else:
+            return Arith(sequences=[*self.sequences, *other.sequences], operation="+")
+
+    def __mul__(self, other):
+        from sequence.sequences.compose.arith import Arith
+
+        if not isinstance(other, Arith) or self._op != other._op:
+            return Arith(sequences=[*self.sequences, other], operation="*")
+        else:
+            return Arith(sequences=[*self.sequences, *other.sequences], operation="*")
+
+    def _resolve_operation(self, operation):
+        """Coerce operation to a numpy.ufunc."""
+        alias_dict = {
+            None: np.mean,
+            "mean": np.mean,
+            "+": np.add,
+            "add": np.add,
+            "*": np.multiply,
+            "mult": np.multiply,
+            "multiply": np.multiply,
+            "min": np.min,
+            "max": np.max,
+        }
+
+        if operation in alias_dict.keys():
+            return alias_dict[operation]
+        else:
+            return operation
+
+    def is_finite(self) -> bool:
+        return any(sequence.is_finite() for sequence in self.sequences)
+
+    def _as_generator(self):
+        op = self._op
+        while True:
+            yield op([next(sequence) for sequence in self.sequences])
+
+    def _as_list(self, stop, start=None, step=None):
+        op = self._op
+        lists = [sequence._as_list(stop, start, step) for sequence in self.sequences]
+        return [op(z) for z in zip(*lists)]
+
+    def _at(self, index: int) -> int:
```

Comment:
the return type is wrong. if you take the mean of the constant sequence 1 with itself, then the resulting sequence is the constant sequence 1/2. Therefore, if you apply the method _at, the return type will be 1/2.

Reply:
> if you take the mean of the constant sequence 1 with itself, then the resulting sequence is the constant sequence 1/2. 

I don't see that, and I must be stupid for not seeing that.

Can you please carry out the computation?

Reply:
here's my computation:

we consider the pointwise mean, so each value in the constant sequence `ArithOp([one, one], "mean")` will have the value that is the mean of two ones.

Substituting definitions for the mean $\mu$ of two numbers $x_1=1$ and $x_2=1$, we have $\mu = \frac{1}{n}\sum_{i=1}^n x_i$, where $n=2$, so, $\mu = \frac{1}{2} (1 + 1) = 2/2 = 1$. Not 1/2.

Reply:
haha sorry :-) I'm the stupid one...

You are right. Let's find a better example:
Suppose a_n is the constant sequence = 1 and b_n is the constant sequence = 2. Then 1/2*(1+2) = 3/2

But this is always the question if we should restrict ourself to integer sequences or not

Reply:
I don't see any scope explosion from allowing real sequences too (in terms of software complexity) - so, why not?

I agree of course that the mean is not a good choice for a default, or an option at all.